test/models/ping_thing_test.rb

```diff
@@ -44,4 +44,52 @@ class PingThingTest < ActiveSupport::TestCase
     refute pt.valid?
     assert_equal ["Signature is too short (minimum is 64 characters)"], pt.errors.full_messages
   end
+
+  test "commitment_level is being correctly assigned and returns correct values" do
+    pt = build(:ping_thing, user_id: @user.id)
```

Comment:
Do you need to assing to specific user in this test? There should be user in factory already

Reply:
Yes, looks like when you build sth associated record is not being created so the user_id is nil.